lib/coradoc/document/title.rb

```diff
@@ -8,12 +8,19 @@ def initialize(content, level, options = {})
         @content = content.to_s
         @id = options.fetch(:id, nil).to_s
         @line_break = options.fetch(:line_break, "")
+        @anchor = options.fetch(:anchor, nil)
       end
 
       def level
         @level ||= level_from_string
       end
 
+      def to_adoc
+        content = Coradoc::Generator.gen_adoc(@content)
```

Comment:
I want to differentiate `@content` being `String` content or Coradoc objects.

Reply:
differentiate in what way? @content can be both strong or Coradoc object because Coradoc::Generator.gen_adoc takes care of Coradoc objects, arrays, strings, nil

Reply:
I don't like the ambiguity of taking into different classes into this method. Can we use different methods for different types of input?

Reply:
Yes, goal would be for each method to be able to check if argument has one specific class?

About the class Title, now that I think about it, @content should not be a string but TextElement instead.

Reply:
It should be a TextElement, you’re right, because it can contain inline formatting like bold/italics and sup/sub etc.